### PR TITLE
Chore remove section split node event

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: sn/src/routing/core/messaging.rs [hunk 1]
```diff
@@ -280,33 +282,76 @@ impl Core {
...
+        });
+
+        match self
+            .send_direct_message_to_nodes(recipients, data_update_msg, prefix.name(), target_pk)
+            .await
+        {
+            Ok(cmd) => Ok(vec![cmd]),
+            Err(err) => {
+                error!(
+                    "Failed to send AE update to our promoted sibling elders: {:?}",
```
Comment (maqi) [inline sn/src/routing/core/messaging.rs:350]: "Failed to send DataUpdate to peers {:?} with error {:?}" ,  recipients, err

File: sn/src/routing/core/mod.rs [hunk 2]
```diff
@@ -280,24 +280,43 @@ impl Core {
 
             // During the split, sibling's SAP could be unknown to us yet.
             // Hence, fire the SectionSplit event whenever detect a prefix change.
+            // We also need to update other nodes w/ our known data.
             let event = if new.prefix != old.prefix {
                 info!("{}: {:?}", LogMarker::SplitSuccess, new.prefix);
                 // In case of split, send AE-Update to sibling new elder nodes.
                 // TODO: confirm in case of sibling's SAP is missing, how to handle such case.
-                commands.extend(self.send_ae_update_to_sibling_section(&old).await);
+                commands.extend(self.send_updates_to_sibling_section(&old).await?);
```
[the remark this answers is not in the document]
Comment (joshuef) [reply sn/src/routing/core/mod.rs:288]: that happens within that function

File: sn/src/node/node_api/handle.rs [hunk 3]
```diff
@@ -102,38 +80,14 @@ impl Node {
...
             }
             NodeDuty::SectionSplit {
-                our_key,
-                our_prefix,
-                our_new_elders,
-                newbie,
+                our_key, newbie, ..
             } => {
-                debug!(
-                    "@@@@@@ SPLIT: Our prefix: {:?}, neighbour: {:?}",
```
Comment (maqi) [inline sn/src/node/node_api/handle.rs:111]: I find this trace log quite useful to track down a valid split.
can we retain it?

File: sn/src/routing/core/mod.rs [hunk 4]
```diff
@@ -297,7 +305,22 @@ impl Core {
...
-            self.send_event(event).await;
+            commands.extend(
+                self.send_data_updates_to(
+                    self.network_knowledge.prefix().await,
+                    self.network_knowledge
+                        .authority_provider()
+                        .await
+                        .peers()
+                        .into_iter()
+                        .filter(|peer| new.elders.contains(&peer.name()))
```
Comment (maqi) [inline sn/src/routing/core/mod.rs:316]: This `self.network_knowledge` is holding `new SAP` as I understand?
Hence, it shall be:  `.filter(|peer| !old.elders.contains(&peer.name()))`  ?

File: sn/src/routing/core/mod.rs [hunk 5]
```diff
@@ -280,11 +280,19 @@ impl Core {
...
-                // In case of split, send AE-Update to sibling new elder nodes.
-                // TODO: confirm in case of sibling's SAP is missing, how to handle such case.
-                commands.extend(self.send_ae_update_to_sibling_section(&old).await);
+                commands.extend(self.send_updates_to_sibling_section(&old).await?);
+                self.retain_members_only(
+                    self.network_knowledge
+                        .members()
+                        .all_members()
+                        .iter()
+                        .map(|peer| peer.name())
```
[the remark this answers is not in the document]
Comment (joshuef) [reply sn/src/routing/core/mod.rs:292]: Should be I think? We only want what's in our section. That's everything

File: sn/src/node/node_api/handle.rs (hunk 3 above)
Comment (joshuef) [reply sn/src/node/node_api/handle.rs:111]: We have `LogMarker::SplitSuccess` for this

File: sn/src/routing/core/mod.rs (hunk 4 above)
Comment (joshuef) [reply sn/src/routing/core/mod.rs:316]: ahhh, yeh makes sense

File: sn/src/node/node_api/handle.rs (hunk 3 above)
Comment (maqi) [reply sn/src/node/node_api/handle.rs:111]: ok then

File: sn/src/routing/core/mod.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (maqi) [reply sn/src/routing/core/mod.rs:292]: but previously the elder.meta only updates `adults` . 
I thought it only for `data holders`
if not, then shall be fine.

File: sn/src/routing/core/mod.rs [hunk 6]
```diff
@@ -280,11 +280,19 @@ impl Core {
 
             // During the split, sibling's SAP could be unknown to us yet.
             // Hence, fire the SectionSplit event whenever detect a prefix change.
-            let event = if new.prefix != old.prefix {
+            // We also need to update other nodes w/ our known data.
+            let event = if (new.prefix != old.prefix) && new.is_elder {
```
Comment (maqi) [inline sn/src/routing/core/mod.rs:284]: this extra condition of new.is_elder will mute the log of SplitSucess among adults?

Comment (joshuef) [reply sn/src/routing/core/mod.rs:284]: yup

**Review by connec — APPROVED** [review]

File: sn/src/routing/core/mod.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (joshuef) [reply sn/src/routing/core/mod.rs:292]: You're right aye sorry! Updated that 👍

**Review by maqi — APPROVED** [review]

**Pull request merged**